[CI] FullClusterRestartIT testWatcherWithApiKey {cluster=UPGRADED} failing

**Build Scans:**
- [elasticsearch-periodic #6880 / openjdk21_8.16.6_java-matrix-bwc](https://gradle-enterprise.elastic.co/s/z5pct3cac3i3w)
- [elasticsearch-pull-request #61322 / 9.1.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/zmrdi7vh5c63g)
- [elasticsearch-pull-request #61293 / 9.1.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/5doc2uonxrxky)

**Reproduction Line:**
```
./gradlew ":x-pack:qa:full-cluster-restart:v8.16.6#bwcTest" -Dtests.class="org.elasticsearch.xpack.restart.FullClusterRestartIT" -Dtests.method="testWatcherWithApiKey {cluster=UPGRADED}" -Dtests.seed=24DA0A14101A2BCC -Dtests.bwc=true -Dtests.locale=sms-FI -Dtests.timezone=America/Anguilla -Druntime.java=21
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.restart.FullClusterRestartIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testWatcherWithApiKey%20%7Bcluster%3DUPGRADED%7D),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: version increased: [true], executed: [false]
Expected: is <true>
     but: was <false>
```

**Issue Reasons:**
- [main] 3 failures in test testWatcherWithApiKey {cluster=UPGRADED} (0.4% fail rate in 830 executions)
- [main] 2 failures in step 9.1.0_bwc-snapshots (1.9% fail rate in 106 executions)
- [main] 2 failures in pipeline elasticsearch-pull-request (2.1% fail rate in 95 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -261,9 +261,6 @@ tests:
 - class: org.elasticsearch.env.NodeEnvironmentTests
   method: testIndexCompatibilityChecks
   issue: https://github.com/elastic/elasticsearch/issues/124388
-- class: org.elasticsearch.xpack.restart.FullClusterRestartIT
-  method: testWatcherWithApiKey {cluster=UPGRADED}
-  issue: https://github.com/elastic/elasticsearch/issues/124159
 - class: org.elasticsearch.multiproject.test.CoreWithMultipleProjectsClientYamlTestSuiteIT
   method: test {yaml=data_stream/190_failure_store_redirection/Redirect ingest failure in data stream to failure store}
   issue: https://github.com/elastic/elasticsearch/issues/124518

